### PR TITLE
#17: [parser] Only innermost groups should be matched by the RE (closes #17)

### LINKED ISSUE #17: [parser] Only innermost groups should be matched by the RE
## description

Only innermost groups should be matched by the RE. There should be no ambiguity.
Currently in `{{consultationLink}}">{{consultationReason}}` `consultationLink}}">{{consultationReason` is matched by the RE.
## how to reproduce

Try write mail with `{{consultationLink}}">{{consultationReason}}`.

### PR DESCRIPTION
**Pull request by calippo**
Closes #17

### tests performed
- old tests should work
- added new unit test

### CODE REVIEW

**Comment by calippo** [comment]
@bytecodeguru codacy was ok, not sure why it failed

File: README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 - Store your email templates where you like.
 - Send your email with the ESP you like.
 - Compose your template using HTML partials (convenient for footers and headers).
+  - Allowed chars in the names of the partials are given by the following regex `a-zA-Z0-9_.-_`
```
Comment (bytecodeguru) [inline README.md]: There is a repeated `_` in the pattern.

**Review by bytecodeguru — CHANGES_REQUESTED** [review]
LGTM, just a little typo in the readme.

**Comment by calippo** [comment]
@bytecodeguru :+1:

**Review by bytecodeguru — APPROVED** [review]
💯

**Pull request merged**